sympy/printing/pretty/pretty.py

```diff
@@ -1154,10 +1154,16 @@ def _print_SingularityFunction(self, e):
             return self._print_Function(e)
 
     def _print_gamma(self, e):
+        from sympy import Function
+        from sympy.functions.special import gamma_functions
+        from sympy import factorial
         if self._use_unicode:
             pform = self._print(e.args[0])
             pform = prettyForm(*pform.parens())
-            pform = prettyForm(*pform.left(greek_unicode['Gamma']))
+            if e.rewrite(factorial) == factorial(e.args[0] - 1):
```

Comment:
If `gamma` is only in the global namespace, then this will not work since the functions are the same. But if it is imported locally from gamma_functions for this comparison, then this should work. It could probably even be `if e.func is gamma`, which more efficient than `==`.